api/dataset.go

```diff
@@ -177,17 +189,17 @@ func (api *DatasetAPI) callDatasetAPI(ctx context.Context, method, path string,
 		}
 	}()
 
-	logData["httpCode"] = resp.StatusCode
-	if resp.StatusCode < http.StatusOK || resp.StatusCode >= 300 {
-		return nil, resp.StatusCode, ErrUnexpectedStatusCode
-	}
-
 	jsonBody, err := ioutil.ReadAll(resp.Body)
 	if err != nil {
 		log.ErrorC("failed to read body from dataset api", err, logData)
 		return nil, resp.StatusCode, err
 	}
 
+	logData["httpCode"] = resp.StatusCode
```

Comment:
The check of the response code has been moved from before reading the response body to after. Don't we want to check the response code first?

Reply:
yes